### DISCUSSION TITLE
Just some questions

### DISCUSSION THREAD

**Discussion by SebastianStehle** [egil/AngleSharp.Diffing#24] [Q&A] [answered]
I have two questions:

1. How are conditional comments compared?

I get some differences about comments and I do not understand it.

Lets say you have a comment like this: 

Is it one big comment or 2 comments and 1 node?

```
<!--[if mso | IE]>
<table />
<![endif]-->
```

If it is 2 comments and 1 node, how are whitespaces handled?

2. Can I ignore some attributes? I would like to ignore empty style attributes like `style=""`

**Comment by SebastianStehle** [comment, marked as answer]
About Question 2

I found it myself. It was stupid. I can just filter the results of course.

**Comment by SebastianStehle** [comment]
I also have a solution for the first problem. I just convert the start and end of conditional comments to normal comments before I compare:

```
// We use a lot of conditional comments in html. These comments are treated as comments, which makes them hard to diff.
            // If we replace them as normal comments the inner html become normal nodes and we can compare them.
            static string Cleanup(string source)
            {
                var regex = new Regex(@"<!--\d{0,}\[(.*)\]\d{0,}>");

                // Replace ending conditional tag to normal comment.
                source = source.Replace("<![endif]-->", "<!-- [endif] -->", StringComparison.OrdinalIgnoreCase);

                // Replace start condition tag to normal comment
                source = regex.Replace(source, x => $"<!-- [{x.Groups[1].Value}] -->");

                return source;
            }
```

**Reply by egil** [reply]
OK cool. What you can do is create a custom diffing strategy, e.g. one that filters them out if they are not relevant for your code, or a custom comparer strategy that handles the normalization for you during comparison, but doesn't change the source or control markup.

Learn more here: https://github.com/AngleSharp/AngleSharp.Diffing/wiki/Creating-Custom-Diffing-Strategies

**Reply by SebastianStehle** [reply]
I tried that, but it did not work. I don't remember why exactly.

**Reply by egil** [reply]
maybe its related to the bug you reported: https://github.com/AngleSharp/AngleSharp.Diffing/issues/25

**Reply by SebastianStehle** [reply]
Perhaps it was just the wrong order in which I registered things
